openapi-core/src/main/java/io/ballerina/openapi/core/generators/common/OASModifier.java

```diff
@@ -973,7 +973,7 @@ private static Optional<String> getCommonPath(List<String> pathValues) {
                 break;
             }
         }
-        return commonPath.isEmpty() ? Optional.empty() : Optional.of(commonPath);
+        return (commonPath.isEmpty() || commonPath.trim().equals(SLASH)) ? Optional.empty() : Optional.of(commonPath);
```

Comment:
Can commonPath have just spaces? In this case this will not return empty

Reply:
Generally the path segments will not have a space. But still it is a valid path. In that case the common path should be with those spaces. So we just need to check whether commonPath is empty or just a slash:
```suggestion
        return (commonPath.isEmpty() || commonPath.equals(SLASH)) ? Optional.empty() : Optional.of(commonPath);
```